### PR TITLE
NFC Move some logic from pyodide.ts to api.ts

### PR DESCRIPTION
**Pull request by hoodmane**
I think this is a bit tidier.

### CODE REVIEW

**Comment by ryanking13** [comment]
Hmm, it is strange that the core test fails in GHA but not in Circle CI.

File: src/js/types.ts [hunk 1]
```diff
@@ -238,8 +238,13 @@ export interface Module {
   noWasmDecoding: boolean;
   quit: (status: number, toThrow: Error) => void;
   preRun: { (): void }[];
-  print: (a: string) => void;
-  printErr: (a: string) => void;
+  print?: (a: string) => void;
+  printErr?: (a: string) => void;
+  arguments: string[];
+  API: API;
```
Comment (ryanking13) [inline src/js/types.ts:244]: Maybe we need to separate methods that comes from emscripten and methods that are added by Pyodide ( no action required for this PR ).

**Review by ryanking13 — APPROVED** [review]
Thanks!

File: src/js/types.ts (hunk 1 above)
Comment (hoodmane) [reply src/js/types.ts:244]: Also, we should move away from Pyodide adding things as much as possible.

**Pull request merged**